File: rb/features.rb
# FreeToPlayGames SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module FreeToPlayGamesFeatures
  def self.make_feature(name)
    case name
    when "base"
      FreeToPlayGamesBaseFeature.new
    when "test"
      FreeToPlayGamesTestFeature.new
    else
      FreeToPlayGamesBaseFeature.new
    end
  end
end

File: py/features.py
# FreeToPlayGames SDK feature factory

from feature.base_feature import FreeToPlayGamesBaseFeature
from feature.test_feature import FreeToPlayGamesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FreeToPlayGamesBaseFeature(),
        "test": lambda: FreeToPlayGamesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
